### ISSUE TITLE
Questions about training the model using CPDMI data

### ISSUE THREAD

**Issue by NianzhenGu** [instanseg/instanseg#43]
Hi! I am trying to train the model using the CPDMI data provided in the notebook `load_datasets.ipynb`.  However, the number of train, validation, and test data are not the same as you mentioned in the paper. I got 114/29/10 (Tr/V/T). 

For the training, I train the model using pixel size 0.5 and try to produce the similar model as` fluorescence_nuclei_and_cells` model. Here is the log. I find several problems:
[experiment_log.csv](https://github.com/user-attachments/files/17598705/experiment_log.csv)
1. The f1 matrix will be 0 for many epochs and reach the plateau quickly.
![f1_metric](https://github.com/user-attachments/assets/894a8274-77d8-4132-a9c9-fdf6a123be42)
This is what the loss looks like:
![loss](https://github.com/user-attachments/assets/9a9dd49d-cb7d-4a88-a97a-40461b1b3a16)
 In the case f1 equals to 0, the output looks like this:
![output_epoch_25](https://github.com/user-attachments/assets/02cbd0c0-5f4e-4d5a-8950-3b0dd9ed8fcd)

3. Some ground truth mask will look like this. The ground truth mask is not proper. 
![output_epoch_70](https://github.com/user-attachments/assets/ecce557c-0f81-49d3-a20b-6d4b8504bfa3)

Thank you very much if you could tell me which part goes wrong. Also, could you please provide the experiment log you used to train the ` fluorescence_nuclei_and_cells` model?

**Comment by ThibautGoldsborough** [comment]
(1) For the plateau, yes that happens sometimes. It's due to lovasz_hinge loss taking forever to converge. You can increase the number of hotstart epochs from 10 to 30 (this substitutes lovasz_hinge with cross entropy for much much faster convergence). Use the --hotstart 30 argument.

(2) This is really strange. It looks like the owners of the dataset may have added new images to their dataset. Did you follow the instructions in our notebook / edit any of our code? This could be a harder  one to debug.

**Comment by ThibautGoldsborough** [comment]
Hey, @walkeraw2002, I know you managed to do this recently. Did you run into this issue? 

Thanks

**Comment by NianzhenGu** [comment]
> (1) For the plateau, yes that happens sometimes. It's due to lovasz_hinge loss taking forever to converge. You can increase the number of hotstart epochs from 10 to 30 (this substitutes lovasz_hinge with cross entropy for much much faster convergence). Use the -hotstart 30 argument.
> 
> (2) This is really strange. It looks like the owners of the dataset may have added new images to their dataset. Did you follow the instructions in our notebook / edit any of our code? This could be a harder one to debug.

Here is what is changed in the `load_CPDMI_Vectra()` function: 1. I change the name to lower case. This is because the dataset I download from Synapse has the lowercase name. 2. Because I'm using windows, I change all "/" to "\\". 
Same for the rest two datasets. The  `load_CPDMI_Vectra()` function already gives me 98/25 images for training and validation. Plus the images from Zeiss, it will exceed the number. 
![image](https://github.com/user-attachments/assets/c6a25b79-b5e2-40e0-ad0f-9c934b0c5bba)
![image](https://github.com/user-attachments/assets/8c37e8b5-33f2-4887-acfb-112272e9f985)

What happened in Zeiss data is the original data (not all) has two tif files ends with `-crop_tif`. The code will detect the image shape and mask shape do not match. But the second one should be the correct original image. So I changed the name of the fist tif image. 
![image](https://github.com/user-attachments/assets/dd7b93cc-f853-4fa5-a445-2aef1cd2a9d6)

**Comment by walkeraw2002** [comment]
> Hey, @walkeraw2002, I know you managed to do this recently. Did you run into this issue?
> 
> Thanks

I did not run into the issue with ground truth cell masks being incorrect. I did not change anything though and just loaded the dataset as specified in the notebook provided, so perhaps it could be that the owners added more images to the dataset. 

The f1 score was 0 for many epochs as well, but it converged faster than it did for Nianzhen:
![image](https://github.com/user-attachments/assets/4caa029a-93cd-4c57-b18c-11e5c39b6e30)

Here is the loss curve from the training as well:
![image](https://github.com/user-attachments/assets/500a35c2-5211-4015-9ae8-4eb57464f5ea)

It seems as though the training loss from my training run was higher than Nianzhen's run, but the testing loss was a bit lower.

I used the following command that was provided by Thibaut:
python train.py -data segmentation_dataset.pth -source "[CPDMI_2023]" --num_epochs 500 --experiment_str fluorescence_nuclei_and_cells -target NC --channel_invariant True --requested_pixel_size 0.5 --augmentation_type heavy

**Comment by NianzhenGu** [comment]
Thanks for answering my question about training! Could you also check the dataset loading notebook based on my previous questions to verify the total number of CPDMI data entries? This is the final number I obtained.
![{E6C842DE-95C6-4880-B068-30C20F10880B}](https://github.com/user-attachments/assets/dbea1f23-3f5c-4b98-9371-b4211dbfe30c)
 
Additionally, there are five hyperparameters that can be adjusted: `mask_threshold`, `mean_threshold`, `overlap_threshold`, `peak_distance`, and `seed_threshold`. Since the documentation is currently unavailable, could you please explain what each of these parameters corresponds to? Thank you!
